### PR TITLE
enable parallel build of kore libraries

### PR DESCRIPTION
**Pull request by dwightguth**
### Scope:

GHC --make supports parallel builds within a single module, but stack and cabal, to the best of my knowledge, do not yet explicitly support this level of parallelism. Thus, we enable parallel builds by default on the components that most benefit from it, by means of a ghc-options flag in kore.cabal.

### Estimate:

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

**Comment by radumereuta** [comment]
On my machine:
Before:
```
[INFO] K Framework Haskell Backend ........................ SUCCESS [03:06 min]
```
After:
```
[INFO] K Framework Haskell Backend ........................ SUCCESS [02:27 min]
```
I will take it.

**Review by radumereuta — APPROVED** [review]
lgtm. Add automerge if you don't want to wait for someone from the haskell backend to review.

**Review by JKTKops — APPROVED** [review]

**Comment by JKTKops** [comment]
This one is simple so I will allow it to automerge (normally I would defer to Ana, but she is out today)

**Pull request merged**
